django/thunderstore/repository/api/experimental/serializers.py

```diff
@@ -95,3 +98,104 @@ class Meta:
             "has_nsfw_content",
             "categories",
         )
+
+
+class PackageUploadAuthorNameField(serializers.SlugRelatedField):
+    """Package upload's author name metadata field."""
+
+    def __init__(self, *args, **kwargs):
+        kwargs["slug_field"] = "name"
+        super().__init__(*args, **kwargs)
+
+    def get_queryset(self):
+        return UploaderIdentity.objects.filter(
+            members__user=self.context["request"].user,
+        )
+
+
+class PackageUploadCategoriesField(serializers.RelatedField):
+    """Package upload's categories metadata field."""
+
+    def get_queryset(self):
+        return PackageCategory.objects.filter(
+            community=self.context["request"].community,
+        )
+
+    def to_representation(self, value):
+        return [c.slug for c in value]
+
+    def to_internal_value(self, data):
+        if not isinstance(data, list):
+            raise serializers.ValidationError("Not a list")
+
+        out = []
+        for category_slug in data:
```

Comment:
```python
categories = self.get_queryset().filter(slug__in=data)
errors = {
    category_slug: f"'{category_slug}' category not found"
    for category_slug in data
    if category_slug not in categories.values_list("slug", flat=True)
}
if errors:
    raise serializers.ValidationError(errors)
return categories
```

In addition to being slightly simpler, it also shows errors for all of the missing categories rather than just the first miss.